### PR TITLE
Debug Helper: Dynamic Notifications Bucket

### LINKED ISSUE #3129: Notifications: Testing Bucket
Let's implement a new URL handler that will help us troubleshoot Notification issues:

```
wordpress://notifications?bucket_name=string
```

When received, the app should map the Notifications bucket as specified. This will allow us to use enterprise builds to retrieve entities stored in [any random] bucket, and test how they are rendered.

### PR DESCRIPTION
**Pull request by jleandroperez**
This is an update to aid debugging of new features:

The app, when launched via `wordpress://notifications?bucket_name=[string]`, will now map the Notifications Bucket as required.

Plus, the custom name will also be displayed in the Notifications Tab's Title.

Closes #3129

/cc @sendhil (Thanks!)

Note: 
Just in case, debug builds scheme is `wpdebug`, while internal beta is `wpinternal`.

### CODE REVIEW

**Comment by sendhil** [comment]
@jleandroperez - in the future can you avoid adjusting spacing and that kind in a targeted PR as it makes it hard to figure out what exactly changed in a PR. Case in point `WordPress/Classes/System/Constants.m` is harder to read because I need to look closely at each line to make sure it's just a spacing thing. @diegoreymendez  has more thoughts on this but going forward I think we're going to do cleanup in separate PRs.

File: WordPress/Classes/System/WordPressAppDelegate.m [hunk 1]
```diff
@@ -1000,6 +1001,19 @@ - (void)logoutSimperiumAndResetNotifications
     [self.simperium signOutAndRemoveLocalData:YES completion:nil];
 }
 
+- (NSString *)notificationsBucketnameFromLaunchOptions:(NSDictionary *)launchOptions
```
Comment (sendhil) [inline WordPress/Classes/System/WordPressAppDelegate.m]: Slight nitpick, but the `name` should probably be capitalized. i.e. `notificationsBucketNameFromLaunchOptions`

File: WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController.m [hunk 2]
```diff
@@ -405,6 +406,19 @@ - (void)showManageButtonIfNeeded
                                                                              action:@selector(showNotificationSettings)];
 }
 
+- (void)showBucketNameIfNeeded
+{
+    // This is only required for debugging:
+    // If we're sync'ing against a custom bucket, we should let the user know about it!
+    Simperium *simperium    = [[WordPressAppDelegate sharedWordPressApplicationDelegate] simperium];
```
Comment (sendhil) [inline WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController.m]: @jleandroperez - didn't we agree to stop the weird spacing on = :-p

Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController.m]: Just maintaining spacing where it already is, but not adding it elsewhere! =D

File: WordPress/Classes/System/WordPressAppDelegate.m (hunk 1 above)
Comment (jleandroperez) [inline WordPress/Classes/System/WordPressAppDelegate.m]: Will change, thank you!

**Comment by jleandroperez** [comment]
@sendhil sorry about the Constants.m change, i just couldn't hold myself. Should i revert that commit?

**Comment by sendhil** [comment]
@jleandroperez - it's fine - in the future would you mind separating it out?

**Comment by sendhil** [comment]
@jleandroperez - mind adjusting the name - but besides that looks good.

**Comment by jleandroperez** [comment]
@sendhil Updated!. Sorry about that, next time, cosmetic adjustments (if any) will go in a different PR.

Mind taking a quick look?. Thanks!

**Comment by sendhil** [comment]
@jleandroperez - looks good. Would you mind addressing the OCLint issues that came up? (one P2 issue)

**Comment by sendhil** [comment]
As we discussed @jleandroperez  - this error was due to the cyclomatic complexity of `WordPressAppDelegate.m` which needs to be refactored badly and isn't your fault. Going to merge this.

**Comment by jleandroperez** [comment]
@sendhil thank you sir!

**Pull request merged**